applications/DEMApplication/python_scripts/sphere_strategy.py

```diff
@@ -391,21 +391,41 @@ def AddAdditionalVariables(self, balls_model_part, DEM_parameters):
     def AddClusterVariables(self, spheres_model_part, DEM_parameters):
         pass
 
-    def AddDofs(self, spheres_model_part=None):
+    # def AddDofs(self, spheres_model_part=None):
+
+    #     if spheres_model_part == None:
+    #         pass
+    #     else:
+    #         for node in spheres_model_part.Nodes:
+    #             node.AddDof(VELOCITY_X)
+    #             node.AddDof(VELOCITY_Y)
+    #             node.AddDof(VELOCITY_Z)
+    #             node.AddDof(ANGULAR_VELOCITY_X)
+    #             node.AddDof(ANGULAR_VELOCITY_Y)
+    #             node.AddDof(ANGULAR_VELOCITY_Z)
+
+    #         Logger.Print("DOFs for the DEM solution added correctly", label="DEM")
+
+    def AddDofs(self):
+        spheres_model_part = self.all_model_parts.Get("SpheresPart")
+        DEM_inlet_model_part = self.all_model_parts.Get("DEMInletPart")
+        cluster_model_part = self.all_model_parts.Get("ClusterPart")
 
-        if spheres_model_part == None:
-            pass
-        else:
-            for node in spheres_model_part.Nodes:
-                node.AddDof(VELOCITY_X)
-                node.AddDof(VELOCITY_Y)
-                node.AddDof(VELOCITY_Z)
-                node.AddDof(ANGULAR_VELOCITY_X)
-                node.AddDof(ANGULAR_VELOCITY_Y)
-                node.AddDof(ANGULAR_VELOCITY_Z)
+        model_part_list = [spheres_model_part, cluster_model_part, DEM_inlet_model_part]
+        for model_part in model_part_list:
+
+            for node in model_part.Nodes:
+                    node.AddDof(VELOCITY_X)
+                    node.AddDof(VELOCITY_Y)
+                    node.AddDof(VELOCITY_Z)
+                    node.AddDof(ANGULAR_VELOCITY_X)
+                    node.AddDof(ANGULAR_VELOCITY_Y)
+                    node.AddDof(ANGULAR_VELOCITY_Z)
```

Comment:
better to use the function in variable_utils for this => [example])(https://github.com/KratosMultiphysics/Kratos/blob/master/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py#L158)

Reply:
Thanks Philip.